### PR TITLE
chore: remove null values from collections/index endpoint

### PR DESCRIPTION
**Pull request by ebezzi**
### Reviewers
**Functional:** 
@MillenniumFalconMechanic 

**Readability:** 
@Bento007 

---

## Changes
- Remove null values from the `collections/index` endpoint. 
- Note: this will also remove null values from other collection related endpoints, but I don't see how this would be a problem.

### CODE REVIEW

**Review by metakuni — APPROVED** [review]
👍

File: tests/unit/backend/corpora/api_server/test_v1_collection.py [hunk 1]
```diff
@@ -769,6 +769,8 @@ def test__get_all_collections_for_index(self):
             visibility=CollectionVisibility.PUBLIC.name,
             owner="test_user_id",
             publisher_metadata={"journal": "Nature"},
+            published_at=datetime.now(),
+            revised_at=datetime.now(),
```
Comment (atolopko-czi) [inline tests/unit/backend/corpora/api_server/test_v1_collection.py:773]: intended change for this PR? (seems out of scope per the PR title)

File: backend/corpora/common/entities/collection.py [hunk 2]
```diff
@@ -116,6 +116,9 @@ def list_attributes_in_time_range(
         def to_dict(db_object):
             _result = {}
             for _field in db_object._fields:
+                _value = getattr(db_object, _field)
+                if _value is None:
+                    continue
```
Comment (atolopko-czi) [inline backend/corpora/common/entities/collection.py:121]: Not sure where else this `to_dict` is used, but it seems like an API-specific change is leaking into ORM-level code. I would consider putting this filtering logic in the API endpoint impl method, for safety.  I think other code that might use this would be surprised if a key/value pair was entirely missing just because the value is None.

File: tests/unit/backend/corpora/api_server/test_v1_collection.py (hunk 1 above)
Comment (atolopko-czi) [inline tests/unit/backend/corpora/api_server/test_v1_collection.py:773]: These props are no longer "required", yet the test is only _now_ asserting their value. I guess I would have expected the test change to be that setting the values to None would cause the response to exclude these props, since that's the nature of the change in the PR.

Comment (ebezzi) [reply tests/unit/backend/corpora/api_server/test_v1_collection.py:773]: Not really since the `null` removal isn't really something we should test, IMHO (it's good to have but it won't break the API contract). These were added now because previously the assertion basically tested two `None` values and therefore passed even without dates (which was arguably wrong). Now they would be removed from the response so the assertion would raise `KeyError` if we checked for these fields.

File: backend/corpora/common/entities/collection.py (hunk 2 above)
Comment (ebezzi) [reply backend/corpora/common/entities/collection.py:121]: Normally I would agree with you but our ORM and API logic is already mixed (for instance, a `to_dict` method shouldn't belong to an ORM at all) and I figured that it would make sense to minimize the amount of data manipulation by putting this additional filter here. I am not particularly opinionated on this though, so I am also happy to do it the way you suggested.

File: tests/unit/backend/corpora/api_server/test_v1_collection.py (hunk 1 above)
Comment (atolopko-czi) [reply tests/unit/backend/corpora/api_server/test_v1_collection.py:773]: Unless we're doing this on all entities consistently, I'd move it to the method impl for which it was added.

File: backend/corpora/common/entities/collection.py (hunk 2 above)
Comment (atolopko-czi) [reply backend/corpora/common/entities/collection.py:121]: Unless we're doing this on all entities consistently, I'd move it to the method impl for which it was added.

Comment (ebezzi) [reply backend/corpora/common/entities/collection.py]: I edited this accordingly.
